examples/hybrid/staggered_nonhydrostatic_model.jl

```diff
@@ -366,39 +407,76 @@ function Wfact!(W, Y, p, dtγ, t)
             error("∂ᶜ𝔼ₜ∂ᶠ𝕄_mode must be :exact when using ρθ")
         end
 
-        # ᶜρθₜ = -ᶜdivᵥ(ᶠinterp(ᶜρθ) * ᶠw)
-        # ∂(ᶜρθₜ)/∂(ᶠw_data) = -ᶜdivᵥ_stencil(ᶠinterp(ᶜρθ) * ᶠw_unit)
-        @. ∂ᶜ𝔼ₜ∂ᶠ𝕄 = -(ᶜdivᵥ_stencil(ᶠinterp(ᶜρθ) * one(ᶠw)))
+        if isnothing(ᶠupwind_product)
+            # ᶜρθₜ = -ᶜdivᵥ(ᶠinterp(ᶜρθ) * ᶠw)
+            # ∂(ᶜρθₜ)/∂(ᶠw_data) = -ᶜdivᵥ_stencil(ᶠinterp(ᶜρθ) * ᶠw_unit)
+            @. ∂ᶜ𝔼ₜ∂ᶠ𝕄 = -(ᶜdivᵥ_stencil(ᶠinterp(ᶜρθ) * one(ᶠw)))
+        else
+            # ᶜρθₜ = -ᶜdivᵥ(ᶠinterp(ᶜρ) * ᶠupwind_product(ᶠw, ᶜρθ / ᶜρ))
+            # ∂(ᶜρθₜ)/∂(ᶠw_data) =
+            #     -ᶜdivᵥ_stencil(
+            #         ᶠinterp(ᶜρ) * ∂(ᶠupwind_product(ᶠw, ᶜρθ / ᶜρ))/∂(ᶠw_data),
+            #     )
```

Comment:
Should we just delete this?

Reply:
(same with commented out code below?)

Reply:
The comments in this function are to help me remember how I derived the Jacobian. I'll get rid of all of them once I have automatic Jacobian constructions, but for now it's probably best to keep them.